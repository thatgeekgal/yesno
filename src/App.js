import React, { Component } from 'react';
import './App.css';
import askApi from './Api';

class App extends Component {
  constructor(props) {
      super(props);
      this.state = { answer: null };
  }

  ask = () => askApi().then(answer => this.setState({ answer }));

  render() {
      return (
          <div className="App">
              <div className="question">
                  <input type="text" />
                  <button type="submit" onClick={this.ask}>Ask to the Gods of the Internet!</button>
              </div>

              {this.state.answer && (
                  <div className="answer">
                      <h1>{this.state.answer.answer}</h1>
                      <img src={this.state.answer.image} alt={this.state.answer.answer} />
                  </div>
              )}
          </div>
      );
  }
}

export default App;