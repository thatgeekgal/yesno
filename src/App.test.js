import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

jest.mock('./Api');

it('should display a text input to fill question', () => {
  const wrapper = shallow(<App />);
  expect(wrapper.find('input[type="text"]').length).toBe(1);
});

it('should display a submit button to send question to the Internet Gods', () => {
  const wrapper = shallow(<App />);
  expect(wrapper.find('button[type="submit"]').length).toBe(1);
});

it ('should not display any answer by default', () => {
  const wrapper = shallow(<App />);
  wrapper.setState({
    answer: null,
  });
  expect(wrapper.find('.answer').length).toBe(0);
});

it('should display Internet Gods answer in both text and picture format', () => {
  const wrapper = shallow(<App />);
  // wrapper.setState({
  //   answer: {
  //       answer: 'no',
  //       forced: false,
  //       image: 'https://yesno.wtf/assets/no/0-b6d3e555af2c09094def76cf2fbddf46.gif',
  //   },
  // });

  wrapper.find('button').simulate('click');

  return Promise.resolve()
    .then(() => {
      wrapper.update();
      const answer = wrapper.find('.answer');
      expect(answer.find('h1').text()).toBe('no');
      expect(answer.find('img').prop('src')).toBe('https://yesno.wtf/assets/no/0-b6d3e555af2c09094def76cf2fbddf46.gif');
    });
});

it('should render <App /> correctly', () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
})