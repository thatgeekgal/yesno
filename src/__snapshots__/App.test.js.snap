// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render <App /> correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="question"
        >
          <input
            type="text"
          />
          <button
            onClick={[Function]}
            type="submit"
          >
            Ask to the Gods of the Internet!
          </button>
        </div>,
        null,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              type="text"
            />,
            <button
              onClick={[Function]}
              type="submit"
            >
              Ask to the Gods of the Internet!
            </button>,
          ],
          "className": "question",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Ask to the Gods of the Internet!",
              "onClick": [Function],
              "type": "submit",
            },
            "ref": null,
            "rendered": "Ask to the Gods of the Internet!",
            "type": "button",
          },
        ],
        "type": "div",
      },
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="question"
          >
            <input
              type="text"
            />
            <button
              onClick={[Function]}
              type="submit"
            >
              Ask to the Gods of the Internet!
            </button>
          </div>,
          null,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                type="text"
              />,
              <button
                onClick={[Function]}
                type="submit"
              >
                Ask to the Gods of the Internet!
              </button>,
            ],
            "className": "question",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Ask to the Gods of the Internet!",
                "onClick": [Function],
                "type": "submit",
              },
              "ref": null,
              "rendered": "Ask to the Gods of the Internet!",
              "type": "button",
            },
          ],
          "type": "div",
        },
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
